### PR TITLE
Add ruby-core CI suite

### PR DESCRIPTION
**Pull request by st0012**
This is similar what [IRB](https://github.com/ruby/irb/blob/master/.github/workflows/ruby-core.yml) and [Reline](https://github.com/ruby/reline/blob/master/.github/workflows/ruby-core.yml) have: a CI suite that runs RDoc changes against Ruby's master branch. It's been very useful to catch issues that would otherwise break CRuby's CI suites.

### CODE REVIEW

**Review by martinemde — COMMENTED** [review]
Makes sense to me. I read the workflow but I haven't seen the others, however, if it runs and passes then I say merge it.

File: .github/workflows/ruby-core.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+name: ruby-core
+
+on:
+  pull_request:
+
+  push:
+    branches:
+      - master
+
+concurrency:
```
Comment (simi) [inline .github/workflows/ruby-core.yml:10]: Any particular reason for this?

**Review by hsbt — APPROVED** [review]
Looks good.

We should consider to move this job as re-using workflow on https://github.com/ruby/actions.

The repositories under the Ruby org only call it.

**Comment by st0012** [comment]
@hsbt Thx, I added it to my todo list 👍

**Pull request merged**